### PR TITLE
Warn to wait for checks if using Dependabot auto-merge

### PR DESCRIPTION
**Pull request by SchulteDev**
Dependabot auto-merge should be used in conjunction with a branch protection rule "Require status checks to pass before merging" for avoiding auto-merge by accident.
When I adapted Dependabot auto-merge, I had been confused about instant auto-merge happening. I would have expected the action to wait for every check to pass (I assumed I didn't understand the dependabotautomerge.yml completely in the first place). Using "Require status checks to pass before merging" is the way to go - highlighting this in the documentation hopefully helps other developers using Dependabot auto-merge correctly in the first run.

### CODE REVIEW

**Comment by cmwilson21** [comment]
@SchulteMarkus Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by mchammer01** [comment]
I'll review this, have just asked someone from the Dependabot Updates team for help.

**Review by AdrianaDJ — COMMENTED** [review]
Create security

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 1]
```diff
@@ -465,6 +465,12 @@ jobs:
 
 If you want to allow maintainers to mark certain pull requests for auto-merge, you can use {% data variables.product.prodname_dotcom %}'s auto-merge functionality. This enables the pull request to be merged when all required tests and approvals are successfully met. For more information on auto-merge, see "[Automatically merging a pull request](/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request)."
```
Comment (felicitymay) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:466]:
```suggestion
If you want to allow maintainers to mark certain pull requests for auto-merge, you can use {% data variables.product.prodname_dotcom %}'s auto-merge functionality. This enables the pull request to be merged when any tests and approvals required by the branch protection rules are successfully met. For more information, see "[Automatically merging a pull request](/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request)" and "[Managing a branch protection rule](/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule)."
```

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 2]
```diff
@@ -465,6 +465,12 @@ jobs:
...
 
+{% warning %}
+
+You have to protect the pull request's target branch to wait for all checks to pass by enabling [Require status checks to pass before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule). Without waiting for the checks, the Dependabot auto-merge will cause an instant merge into the pull request's target branch.
+
+{% endwarning %}
+
```
Comment (felicitymay) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:473]: There are several types of branch protection rules that you might want to enable, in addition to "Require status checks to pass before merging". So I've updated the paragraph above to clarify the expected behavior and I'm going to modify this warning according to our style guide.

```suggestion
{% note %}

**Note:** If you use status checks to test pull requests, you should enable **Require status checks to pass before merging** for the target branch for {% data variables.product.prodname_dependabot %} pull requests. This branch protection rule ensures that pull requests are not merged unless all the required status checks pass. For more information, see "[Managing a branch protection rule](/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule)."

{% endnote %}

```

**Review by felicitymay — COMMENTED** [review]
@SchulteMarkus - many thanks for opening this pull request ✨ 

You make a great point that the original text forgot to mention branch protection rules. After discussion with the subject matter expert, I'm adapting your changes to make the original text clearer, in addition to the new note.

**Comment by felicitymay** [comment]
I'm going to apply my suggestions and check the preview. If all's well, we'll be happy to merge this pull request.

**Review by felicitymay — APPROVED** [review]

**Comment by felicitymay** [comment]
@SchulteMarkus - many thanks for contributing this improvement to the docs. This should save other developers from making this mistake.

**Pull request merged**